crates/sui-core/src/authority/authority_per_epoch_store_pruner.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::authority::authority_per_epoch_store::EPOCH_DB_PREFIX;
+use std::fs;
+use std::path::PathBuf;
+use sui_types::base_types::EpochId;
+use tracing::log::{error, info};
+use typed_store::rocks::safe_drop_db;
+
+pub(crate) async fn prune_old_epoch_data(
+    parent_path: PathBuf,
+    epoch: EpochId,
+    epoch_number_to_retain: usize,
+) {
+    match prune_old_directories(parent_path, epoch, epoch_number_to_retain) {
+        Ok(pruned_count) => info!(
+            "Finished pruning old epoch databases. Pruned {} dbs",
+            pruned_count
+        ),
+        Err(err) => error!("Error while removing old epoch databases {:?}", err),
+    }
+}
+
+fn prune_old_directories(
+    parent_path: PathBuf,
+    epoch: EpochId,
+    epoch_number_to_retain: usize,
+) -> Result<usize, anyhow::Error> {
+    let mut pruned = 0;
+    if epoch_number_to_retain == 0 || epoch_number_to_retain == usize::MAX {
+        return Ok(pruned);
```

Comment:
nit: maybe add a info log here saying, we are skipping pruning because of the setting